howtos/how_to_request_cached_data.py

```diff
@@ -0,0 +1,139 @@
+"""
+This module contains example snippets on how to request cached data from a direct receiver. These examples assume
+that the direct receiver has already been started, and that it was created from a messaging service which has
+already been connected.
+"""
+
+from solace.messaging.utils.correlation_completion_listener import CorrelationCompletionListener
+from solace.messaging.utils.cache_request_outcome import CacheRequestOutcome
+from solace.messaging.resources.topic_subscription import TopicSubscription
+from solace.messaging.resources.cached_message_subscription_request import CachedMessageSubscriptionRequest
+
+class MyCorrelationCompletionListener(CorrelationCompletionListener):
+    def on_completion(result: CacheRequestOutcome, correlation_id: int, e: Exception):
+        print("Completed!")
+
+class HowToUseCachedMessageSubscriptionRequests:
+    """
+    Sampler for making cache requests. Usage of cache subscriptions require
+    configuration and hosting of SolCache application.
+    """
+    def create_subscription_request_to_receive_cached_and_live_messages(subscription_expression: str,
+                                                                        cache_name: str,
+                                                                        cache_access_timeout: int,
+                                                                        direct_message_receiver: 'DirectMessageReceiver'):
+        """
+        An example how to create and perform CachedMessageSubscriptionRequest to receive a
+        mix of live and cached messages matching specified topic subscription.
+        Args:
+            subscription_expression(str): topic subscription expression
+            cache_name(str): name of the solace cache to retrieve from
+            cache_access_timeout(int): solace cache request timeout, in milliseconds
+            direct_message_receiver(DirectMessageReceiver): an already configured and connected receiver
+
+        Raises:
+            InterruptedException: if any thread has interrupted the current thread
+            PubSubPlusClientException: if the operation could not be performed, i.e. when
+            cache request failed and subscription rolled back/removed
+            IllegalStateException: If the service is not running
+        """
+        cached_message_subscription_request = CachedMessageSubscriptionRequest.asAvailable(cache_name,
+                                                                                           TopicSubscription.of(subscription_expression),
+                                                                                           cache_access_timeout)
+        correlation_id = 12345
```

Comment:
Correlation ID "should" be unique, lets not re-use the same one for all of the examples
This applies to all the functions that use this id

Reply:
Please review fix